doc/conf.py

```diff
@@ -222,6 +240,21 @@
     "codebgcolor": "#f5efe7",
 }
 
+
+def generate_example_rst(app, what, name, obj, options, lines):
+    # generate empty examples files, so that we don't get
+    # inclusion errors if there are no examples for a class / module
+    examples_path = os.path.join(app.srcdir, "generated",
+                                 "%s.examples" % name)
+    if not os.path.exists(examples_path):
+        # touch file
+        open(examples_path, 'w').close()
+
+
+def setup(app):
+    app.connect('autodoc-process-docstring', generate_example_rst)
```

Comment:
Is it really needed for the joblib documentation? If it's a generic feature maybe it should be contributed upstream to sphinx-gallery project instead of copy pasting in each project that use sphinx-gallery.